pom.xml

```diff
@@ -107,6 +107,12 @@
             <version>${junit.jupiter.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
```

Comment:
Please do not.

I have nothing but bad experiences with Lombok.
I know that some people may find it useful to write less code, however code that is written behave as expected (vs generation of all possible constructors for ex.).
At best it messes up source jar.

Reply:
I'm happy to take it out, it shouldn't be packaged in the source jar though.  

I've got to finish ttl and then I've done all the policies so I'll probably push it all back as one.

Reply:
>  it shouldn't be packaged in the source jar though

That is not what I meant, rather that sources using Lombok are not valid raw Java.
When navigating sources of a third-party library, IDEs match / compile source code with no knowledge of Lombok (or other tools like it) generating stuff at compile time.

This ends up with broken navigation, (navigating to a constructor that does not exists for instance), line numbers not matching those in stacktraces and inaccurate breakpoints.

Reply:
> I'll probably push it all back as one

If you can split into feature complete (1 commit = prod+test) commits to ease review I would be very grateful !